d2layouts/d2near/layout.go

```diff
@@ -94,8 +150,16 @@ func WithoutConstantNears(ctx context.Context, g *d2graph.Graph) (nears []*d2gra
 		}
 		_, isConst := d2graph.NearConstants[d2graph.Key(obj.Attributes.NearKey)[0]]
 		if isConst {
-			nears = append(nears, obj)
-			g.Objects = append(g.Objects[:i], g.Objects[i+1:]...)
+			descendantObjects, edges := pluckOutNearObjectAndEdges(g, obj)
+
+			tempGraph := d2graph.NewGraph()
+			tempGraph.Root.ChildrenArray = []*d2graph.Object{obj}
+			tempGraph.Root.Children[obj.ID] = obj
+			tempGraph.Objects = descendantObjects
```

Comment:
i think the splitting of graphs has to be more robust. currently it's the bare minimum to get a layout working, but a layout code may change and depend on other parts of the graph structure that are not properly split.

specifically,
- each descendant object has to have all their references changed. the graph reference has to be updated: https://github.com/donglixiaoche/d2/blob/b9dd247c75013d0581d15ae2bb3490ca86e0ba4f/d2graph/d2graph.go#L79

- The parent of the top-level object has to be updated.

- they should be removed from the previous root's children and childrenarray.


and then when it's merged back in, all those should be updated again to be the same as before.

put another way: when the core layout is running on each of these graphs, it should have no way of reaching an object in a separate graph.

Reply:
> i know that's a little ambiguous, but having a clean way to do this will be useful for more than just `near`. later when D2 supports other graph types, we'll want to run different layout algorithms for those too, requiring a clean split

totally argee with you, fixed

Reply:
nice, that looks clean to me.